src/Orleans.Core/CodeGeneration/TypeUtils.cs

```diff
@@ -556,24 +556,45 @@ public static IEnumerable<TypeInfo> GetDefinedTypes(Assembly assembly, Logger lo
             }
             catch (Exception exception)
             {
-                if (logger != null && logger.IsWarning)
-                {
-                    var message =
-                        $"Exception loading types from assembly '{assembly.FullName}': {LogFormatter.PrintException(exception)}.";
-                    logger.Warn(ErrorCode.Loader_TypeLoadError_5, message, exception);
-                }
-                
                 var typeLoadException = exception as ReflectionTypeLoadException;
+
                 if (typeLoadException != null)
                 {
+                    if (typeLoadException.LoaderExceptions != null)
+                    {
+                        //
+                        // If we've only BadImageFormatExceptions in LoaderExceptions, then it's ok to not to log, otherwise log
+                        // as a warning.
+                        //
+
+                        if (logger != null && logger.IsWarning)
```

Comment:
if we do include a property to enable more verbose logging, it would be nice to not skip logging BadImageFormatException, potentially with `Trace` level. But even for our own troubleshooting, I think we should enable that regardless of when we add the msbuild property support

Reply:
CodeGen is not on ILogger yet, but I add the IsVerbose to log it in case of verbose logging set.